src/index.js

```diff
@@ -0,0 +1,227 @@
+const Express = require("express");
+const BodyParser = require("body-parser");
+const Crypto = require("crypto");
+
+const app = Express();
+
+app.use(Express.static("public"));
+app.use(BodyParser.json());
+
+/**
+ * Given that ROCK = 0, PAPER = 1 and SCISSOR = 2
+ * Returns the winning hand (0, 1 or 2) or -1 if tie
+ *
+ * @param a opponent
+ * @param b opponent
+ * @returns {number} winning hand (a or b) or -1 if tied
+ */
+
+const CHOICES = {
+  ROCK: 0,
+  PAPER: 1,
+  SCISSOR: 2,
+};
+
+const COMPARE_RESULT = {
+  DRAW: Symbol("draw"),
+  ...CHOICES,
+};
+const compareHands = (a, b) =>
+  a === b ? COMPARE_RESULT.DRAW : 2 - ((a + b) % 3);
+
+const prb = (bytes) => Crypto.pseudoRandomBytes(bytes).toString("hex");
+
+class Round {
+  #hands = new Map();
+
+  constructor(hands) {
+    if (hands.length !== 2) {
+      throw new Error("Expected exactly 2 hands");
+    }
+
+    this.#hands = new Map(hands.map((id) => [id, null]));
+  }
+
+  static handId(hand) {
+    return hand[0];
+  }
+
+  static handValue(hand) {
+    return hand[1];
+  }
+
+  static handPlayed(hand) {
+    return Round.handValue(hand) != null;
+  }
+
+  static everyHandPlayed(...hands) {
+    return hands.every((hand) => Round.handPlayed(hand));
+  }
+
+  setHand(id, choice) {
+    if (!this.#hands.has(id)) {
+      throw new Error("No such hand");
+    }
+    this.#hands.set(id, Number(choice));
+  }
+
+  #getHands() {
+    return Array.from(this.#hands.entries());
+  }
+
+  canResolve() {
+    const [a, b] = this.#getHands();
+    return Round.everyHandPlayed(a, b);
+  }
+
+  getWinner() {
+    const [a, b] = this.#getHands();
+    if (!Round.everyHandPlayed(a, b)) {
+      return null;
+    }
+
+    const winner = compareHands(Round.handValue(a), Round.handValue(b));
+    if (winner === COMPARE_RESULT.DRAW) {
+      return COMPARE_RESULT.DRAW;
+    }
+
+    return winner === Round.handValue(a) ? Round.handId(a) : Round.handId(b);
+  }
+
+  toJSON() {
+    const winner = this.canResolve() ? this.getWinner() : undefined;
+    return {
+      played: this.#getHands().filter((hand) => Round.handPlayed(hand)).length,
+      result: winner === COMPARE_RESULT.DRAW ? "draw" : winner,
```

Comment:
Move the stringifaction of compare result to a separate function, there are a lot of repetitions if this.